crates/sui-indexer-alt-graphql/src/api/types/epoch.rs

```diff
@@ -205,83 +178,60 @@ impl Epoch {
         scope: Scope,
         epoch_id: Option<UInt53>,
     ) -> Result<Option<Self>, RpcError> {
-        let start = EpochStart::empty(scope)
-            .fetch(ctx, epoch_id.map(|id| id.into()))
-            .await?;
-
-        let Some(contents) = &start.contents else {
-            return Ok(None);
-        };
-
-        Ok(Some(Self {
-            epoch_id: contents.epoch as u64,
-            start,
-        }))
-    }
-}
-
-impl EpochStart {
-    fn empty(scope: Scope) -> Self {
-        Self {
-            scope,
-            contents: None,
-        }
-    }
-
-    /// Attempt to fill the contents. If the contents are already filled, returns a clone,
-    /// otherwise attempts to fetch from the store. The resulting value may still have an empty
-    /// contents field, because it could not be found in the store, or the epoch started after the
-    /// checkpoint being viewed.
-    async fn fetch(&self, ctx: &Context<'_>, epoch_id: Option<u64>) -> Result<Self, RpcError> {
-        if self.contents.is_some() {
-            return Ok(self.clone());
-        }
-
         let pg_loader: &Arc<DataLoader<PgReader>> = ctx.data()?;
 
-        let load = if let Some(id) = epoch_id {
-            pg_loader.load_one(EpochStartKey(id)).await
-        } else {
-            let cp = self.scope.checkpoint_viewed_at();
-            pg_loader.load_one(CheckpointBoundedEpochStartKey(cp)).await
+        let stored = match epoch_id {
+            Some(id) => pg_loader.load_one(EpochStartKey(id.into())).await,
```

Comment:
We can also do `Self::with_id(id).get_start(ctx)` here to instantiate the `Epoch` to keep epoch start fetching logic and the filtering of `checkpoint_viewed_at` in one place. But the duplication is pretty minimal so I don't have strong opinions here.